lib/note-editor.jsx

```diff
@@ -76,19 +71,19 @@ export const NoteEditor = React.createClass( {
 
 	onCancelRevision: function() {
 		// clear out the revision
-		this.setState( { revision: null } );
+		this.props.onRevisionChange( { revision: null } );
 		this.setIsViewingRevisions( false );
 	},
 
-	setIsViewingRevisions: function( isViewing ) {
-		this.setState( { isViewingRevisions: isViewing } );
+	setIsViewingRevisions: function( isViewingRevisions ) {
+		this.props.onRevisionChange( { isViewingRevisions } );
 	},
```

Comment:
coming in here now would be a great opportunity to trim down on the noise…

```js
setIsViewingRevisions = isViewingRevisions => this.props.setRevisionChange( { isViewingRevisions } );
```

also, I'm a bit leery of using an object and key name to call a function that only takes one argument